lib/react_client/js_backed_map.dart

```diff
@@ -0,0 +1,153 @@
+@JS()
+library js_map;
+
+import 'dart:collection';
+import 'dart:core';
+import 'dart:js_util' as js_util;
+
+import 'package:js/js.dart';
+
+/// A view into a JavaScript object ([jsObject]) that conforms to the Dart [Map] interface.
+///
+/// The keys are all enumerable properties of the [jsObject], though non-enumerable properties may also be accessed.
+/// For this reason, it is recommended to use plain, simple JavaScript objects.
+///
+/// Keys should be of primitive types (String, num, bool, null), as any non-primitive Dart objects will be
+/// canonicalized to their JavaScript string representations upon read/write.
+/// For performance reasons, keys are not validated.
+///
+/// `null` is allowed as a key.
+///
+/// Iteration order is arbitrary, and is based on the current browser's implementation.
+///
+/// Two JsBackedMap instances are considered equal if they are backed by the same [jsObject].
+class JsBackedMap extends MapBase<dynamic, dynamic> {
```

Comment:
This implementation was moved, and pretty much just got things removed from it.

However, it could use a thorough re-review.